Sources/TransifexNative/Cache.swift

```diff
@@ -103,73 +119,257 @@ public final class TXDiskCacheProvider: NSObject, TXCacheProvider {
             }
         }
-        super.init()
     }
 }
 
-extension TXProviderBasedCache: TXCache {
-    public func get(key: String, localeCode: String) -> String? {
-        return memoryCache.get(key: key, localeCode: localeCode)
+/// Decorator class responsible for updating the passed internalCache using a certain override policy defined
+/// in the `TXCacheOverridePolicy` enum.
+@objc
+public final class TXStringOverrideFilterCache: TXDecoratorCache {
+    let policy: TXCacheOverridePolicy
+    
+    @objc
+    public init(policy: TXCacheOverridePolicy,
+                internalCache: TXCache) {
+        self.policy = policy
+        super.init(internalCache: internalCache)
     }
     
-    /// Provider based cache doesn't update its internal cache after being initialiazed, so the update
-    /// method of the TXCache protocol is a no-op.
+    override public func update(translations: TXTranslations) {
+        if policy == .overrideAll {
+            super.update(translations: translations)
+            return
+        }
+        
+        var updatedTranslations = self.get()
     
-    public func update(translations: TXTranslations,
-                       replaceEntries: Bool) {
-        /// No-op
+        for (localeCode, localeTranslations) in translations {
+            for (stringKey, translation) in localeTranslations {
+                // Make sure that the new translation has a value and it's not
+                // an empty string.
+                guard let translatedString = translation[TXDecoratorCache.STRING_KEY],
+                      translatedString.count > 0 else {
+                    continue
+                }
+                
+                    // If the policy set set to override untranslated only, then
+                    // update the cache only if there's no existing translation
+                    // for that stringKey.
+                if (policy == .overrideUntranslatedOnly
+                        && self.get(key: stringKey,
+                                localeCode: localeCode) == nil)
+                   ||
+                    // If the policy is set to override using translated only,
+                    // then update the cache only if there's an existing
+                    // translation for that stringKey.
+                    (policy == .overrideUsingTranslatedOnly
+                        && self.get(key: stringKey,
+                                    localeCode: localeCode) != nil) {
+                    if updatedTranslations[localeCode] == nil {
+                        updatedTranslations[localeCode] = [:]
+                    }
+
+                    updatedTranslations[localeCode]?[stringKey] = translation
+                }
+            }
+        }
+
+        super.update(translations: updatedTranslations)
     }
 }
 
+/// The standard wrapper cache that the TxNative SDK is initialized with, if no other cache is provided.
```

Comment:
I would remove `wrapper` from here, I don't think it provides any useful context. I would go with `The default cache` or `The standard cache`.

Reply:
Fixed